src/channels/CredentialChannels/backgroundCredentialChannel.ts

```diff
@@ -37,6 +38,16 @@ export async function showCredentialPopup(
         // @ts-expect-error compatibility with old cType interface
         content.cTypes = content.cTypes?.map((cType) => cType.schema);
       }
+
+      // the DID to use for signing could be predetermined by the dApp, if we have a matching identity we’ll use it
+      const { owner } = content.claim;
+      const identities = await getIdentities();
+      for (const { address, did } of Object.values(identities)) {
+        if (did === owner) {
+          await setCurrentIdentity(address);
+        }
+      }
```

Comment:
Does the loop need to continue after the matching identity is found?

Reply:
No, but it doesn’t really hurt, while completely avoiding the concerns about the flow control